Remove anys and type assertions (#405)

pshycodr opened the pull request:
#405 

## Description

Removed all instances of `any` from the codebase, expect the `error` types where it is seems necessary.

sodic commented:
Hey @pshycodr. Thanks for the contribution.

From what I can tell, this PR does more than just remove the `as` and `any`s. So I'll kindly ask you to stick to the original issue to minimize the change surface.

A good rule of thumb is to avoid mixing refactors and formatting with other changes. See this blog post for all the tips on good PRs: https://mtlynch.io/code-review-love/

Finally, the PR seems to have some weird leftovers. Was this LLM generated, and if so, did you double check the code?

pshycodr commented:
@sodic 
apologies for the earlier changes. I have  now updated the PR to strictly follow the original issue. and removed all the major code structure and logic changes.
Let me know if any further changes are needed!

sodic commented:
Hey @pshycodr, thanks for the contribution.

Unfortunatley, I don't have time to dig into this PR at the moment.

It seems like the bulk of it was LLM generated (strange comment leftovers, new unexplained type assertions, etc.). I can't trust it enough to merge it without a detailed check. On the other hand, detailed check will take some time and the issue is not important enough to justify it.

My apologies for not saying this right away.

For future issues, please check whether we're accepting outside contributor help before starting to work on them (for example, first comment on the issue and only start the PR after we give you a go-ahead). This PR was a bad fit for outside contributors (e.g., we didn't necessarily just want to add types to untyped values, we potentially wanted to improve them too).

I'm closing for now but promise to give you credit if we end up using your contribution in the future.

pshycodr commented:
Totally understood, and thanks for the honest response.
Apologies again for not confirming before picking up the issue,  I’ll make sure to align first going forward.
Appreciate the learning and the opportunity to engage , and if any part of the PR helps down the line, I’m glad it added value.
Looking forward to contributing again, the right way this time.
Cheers.
